master/internal/grpcutil/auth.go

```diff
@@ -157,7 +158,7 @@ func GetUser(ctx context.Context) (*model.User, *model.UserSession, error) {
 			return nil, nil, ErrPermissionDenied
 		}
 		return userModel, session, nil
-	case sql.ErrNoRows, db.ErrNotFound:
+	case sql.ErrNoRows, db.ErrNotFound, jwt.ErrTokenExpired:
```

Comment:
for my understanding, why did we need this? looking at `UserByToken`, i would think the `v2.Verify` call implodes already and you get a `db.ErrNotFound` for expires.

Reply:
yes, i understand that. when i say 'db error' i mean this ~~dumb~~ wonderful place where we swallow the verification error and return a db error https://github.com/determined-ai/determined/blob/36dac4ba5c3516c978c246702659446b886e97f0/master/internal/user/postgres_users.go#L126

Reply:
but i see now this probably branches before that due to the external token with saas.

Reply:
but overall, i was just trying to understand why this change needs this extra case, or it is a bug fix/only tangentially related.

Reply:
Oh this is actually in connection with a related bug in MLDE code, which is being discussed by @billboggs. Part of reason is that the `jwt` package has some unusual handling of it's own error types - bit of a trade-off between making sure we return the correct error code so not everything looks like a server error and authentication problems can be handled appropriately, and wanting to keep implementation details of UserByExternalToken from leaking out.